testing/trino-testing/src/main/java/io/trino/testing/PlanDeterminismChecker.java

```diff
@@ -47,6 +47,7 @@ public void checkPlanIsDeterministic(String sql)
         checkPlanIsDeterministic(localQueryRunner.getDefaultSession(), sql);
     }
 
+    @SuppressWarnings("ReturnValueIgnored") // Stream#reduce called for side effects (assertions)
```

Comment:
I don't think obsolete/inactive suppressions cause errors or even warnings (this can be difficult to do when each static analysis tool has its own and the other analyzers don't know about them) and I agree that would be ideal.

I also agree that a `for`loop should be used when the expected result is side effects. So I converted this to a loop, and now I'm worried that the loop bounds aren't right :) `IntStream.range(1, MINIMUM_SUBSEQUENT_SAME_PLANS)` should have been `rangeClosed` perhaps? Or iterating from `0`?